src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -474,35 +537,61 @@ int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
      */
     int encryptFinal(byte[] in, int inOfs, int len, byte[] out, int outOfs)
         throws IllegalBlockSizeException, ShortBufferException {
-        if (len > MAX_BUF_SIZE - tagLenBytes) {
-            throw new ShortBufferException
-                ("Can't fit both data and tag into one buffer");
-        }
+        checkDataLength(processed, Math.addExact(len, tagLenBytes));
+
```

Comment:
Now that encrypt(byte[], int, int, byte[], int) may also store data into 'ibuffer', shouldn't this encryptFinal() method processes bytes in 'ibuffer' before processing 'in'? The check here would also needs to be updated with ibuffer.size()?

Reply:
This is not an optimized path for bytebuffers and will never have any data in ibuffer.

Reply:
Hmm, ok, can we add a check on ibuffer size here just to be sure?

Reply:
I've rewritten checkDataLength so it can include more values to calculate.